server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -413,8 +413,14 @@ public void addFinishedBlockIds(
               + " bitmaps!");
     }
 
+    ShuffleTaskInfo taskInfo = getShuffleTaskInfo(appId);
+    if (taskInfo == null) {
+      throw new InvalidRequestException(
+          "ShuffleTaskInfo is not found that should not happen for appId: " + appId);
+    }
     for (Map.Entry<Integer, long[]> entry : partitionToBlockIds.entrySet()) {
       Integer partitionId = entry.getKey();
+      taskInfo.incrBlockNumber(shuffleId, partitionId, entry.getValue().length);
```

Comment:
Would we ever see a report shuffle result request being retried (as in GRPC, not a stage retry)? If so, we would get in trouble here.

Reply:
Updated. The retried request will be filtered.

Reply:
how? where?

Reply:
If the blockId exists in the bitmap, the blockId counter increment will be ignored. This may need to log if having any duplicate blockId occur

Reply:
got it